### PR TITLE
Improve animation + clipping

### PR DESCRIPTION
**Pull request by agirault**

### CODE REVIEW

File: Sources/Proxy/Core/AbstractRepresentationProxy/index.js [hunk 1]
```diff
@@ -302,7 +302,7 @@ function extend(publicAPI, model, initialValues = {}) {
   Object.assign(model, DEFAULT_VALUES, initialValues);
 
   macro.obj(publicAPI, model);
-  macro.get(publicAPI, model, ['input', 'actors', 'volumes']);
+  macro.get(publicAPI, model, ['input', 'mapper', 'actors', 'volumes']);
```
Comment (jourdain) [inline Sources/Proxy/Core/AbstractRepresentationProxy/index.js]: But if there is no mapper?

Comment (jourdain) [reply Sources/Proxy/Core/AbstractRepresentationProxy/index.js]: Meaning `model.mapper` is `undefined` as it is not used or set?

Comment (agirault) [reply Sources/Proxy/Core/AbstractRepresentationProxy/index.js]: Can a representation not have a mapper?? and if it is undefined won't that just return undefined?

File: Sources/Proxy/Core/ViewProxy/index.js [hunk 2]
```diff
@@ -253,9 +253,9 @@ function vtkViewProxy(publicAPI, model) {
 
   publicAPI.setAnimation = (enable) => {
     if (enable) {
-      model.renderWindow.getInteractor().requestAnimation('proxy');
+      model.renderWindow.getInteractor().requestAnimation(publicAPI);
     } else {
-      model.renderWindow.getInteractor().cancelAnimation('proxy');
+      model.renderWindow.getInteractor().cancelAnimation(publicAPI);
```
Comment (agirault) [inline Sources/Proxy/Core/ViewProxy/index.js]: @jourdain is that enough to make distinction between different viewproxys?

File: Sources/Rendering/Core/ImageMapper/index.js [hunk 3]
```diff
@@ -113,6 +113,30 @@ function vtkImageMapper(publicAPI, model) {
     return image.extentToBounds(ex);
   };
 
+  publicAPI.getBoundsForSlice = (slice, thickness = 0) => {
```
Comment (agirault) [inline Sources/Rendering/Core/ImageMapper/index.js]: @martinken @jourdain I need this on the ImageSlice also to my understanding, to apply the necessary transform

File: Sources/Proxy/Core/AbstractRepresentationProxy/index.js (hunk 1 above)
Comment (jourdain) [reply Sources/Proxy/Core/AbstractRepresentationProxy/index.js]: yes a rep can be anything. But I guess that's ok to return undefined. 
Just wanted to mention that you should not expect to always have a mapper in a rep.

Comment (agirault) [reply Sources/Proxy/Core/AbstractRepresentationProxy/index.js]: don't representations always have a mapper? And even if it didn't, wouldn't it just return `undefined` and then whoever called this need to handle it?

File: Sources/Proxy/Core/ViewProxy/index.js (hunk 2 above)
Comment (jourdain) [reply Sources/Proxy/Core/ViewProxy/index.js]: yes I guess.

File: Sources/Rendering/Core/ImageMapper/index.js (hunk 3 above)
Comment (jourdain) [reply Sources/Rendering/Core/ImageMapper/index.js]: ok

File: Sources/Rendering/Core/RenderWindowInteractor/index.js [hunk 4]
```diff
@@ -604,7 +632,7 @@ function vtkRenderWindowInteractor(publicAPI, model) {
   // do not want extra renders as the make the apparent interaction
   // rate slower.
   publicAPI.render = () => {
-    if (model.requestAnimationCount === 0) {
+    if (model.animationRequest === null) {
```
Comment (jourdain) [inline Sources/Rendering/Core/RenderWindowInteractor/index.js]: are you sure you want to compare against null?

Comment (agirault) [reply Sources/Rendering/Core/RenderWindowInteractor/index.js]: :man_shrugging:

Comment (agirault) [reply Sources/Rendering/Core/RenderWindowInteractor/index.js]: Could also do `animationRequesters.size === 0`

Comment (jourdain) [reply Sources/Rendering/Core/RenderWindowInteractor/index.js]: I don't know but just wanted to mention that an object can be `null`, `undefined` and `undefined === null => false`

Comment (agirault) [reply Sources/Rendering/Core/RenderWindowInteractor/index.js]: I see. Then should be all good:
- init: https://github.com/Kitware/vtk-js/pull/575/files/3b786ece3a26aec42bba6ede6ee8768b6754a21f#diff-f698a8e6250bde65d1ae6422124e9a84R870
- deleted: https://github.com/Kitware/vtk-js/pull/575/files/3b786ece3a26aec42bba6ede6ee8768b6754a21f#diff-f698a8e6250bde65d1ae6422124e9a84R291

File: Sources/Rendering/Core/ImageMapper/index.js (hunk 3 above)
Comment (agirault) [reply Sources/Rendering/Core/ImageMapper/index.js]: I don't quite understand how that is working here... https://github.com/Kitware/vtk-js/blob/master/Sources/Rendering/Core/ImageSlice/index.js#L56-L102

Comment (agirault) [reply Sources/Rendering/Core/ImageMapper/index.js]: done, ready

File: Sources/Proxy/Core/AbstractRepresentationProxy/index.js (hunk 1 above)
Comment (agirault) [reply Sources/Proxy/Core/AbstractRepresentationProxy/index.js]: your call, i don't need it anymore for my use case (and could access it by reference instead than get)

**Pull request merged**